### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by ik-web**

### CODE REVIEW

File: src/inverseRobot.js [hunk 1]
```diff
@@ -24,7 +24,16 @@
  * @return {object}
  */
 function inverseRobot(robot) {
-  // write code here
+  for (const key in robot) {
+    if (!robot[robot[key]]) {
```
Comment (illiarerun) [inline src/inverseRobot.js:28]: [Checklist](https://github.com/mate-academy/js_inverse_robot/blob/master/checklist.md) 2nd item, what if value contains falsy value? empty string, null, false, etc. Try to fix your check

File: src/inverseRobot.js [hunk 2]
```diff
@@ -24,7 +24,16 @@
...
+      robot[robot[key]] = key;
+      delete robot[key];
+    } else {
+      return null;
+    }
```
Comment (illiarerun) [inline src/inverseRobot.js:33]: You can invert your condition to avoid using `else` construction

File: src/inverseRobot.js [hunk 3]
```diff
@@ -24,7 +24,16 @@
  * @return {object}
  */
 function inverseRobot(robot) {
-  // write code here
+  for (const key in robot) {
+    if (!robot[robot[key]]) {
+      robot[robot[key]] = key;
+      delete robot[key];
```
Comment (illiarerun) [inline src/inverseRobot.js:30]: you should never modify function arguments!
And also this action is redundant
```suggestion
```

**Review by illiarerun — CHANGES_REQUESTED** [review]
Nice star, now let's work on fixes:

File: src/inverseRobot.js (hunk 1 above)
Comment (ik-web) [reply src/inverseRobot.js]: Fixed

File: src/inverseRobot.js (hunk 2 above)
Comment (ik-web) [reply src/inverseRobot.js]: Fixed

File: src/inverseRobot.js [hunk 4]
```diff
@@ -24,7 +24,17 @@
  * @return {object}
  */
 function inverseRobot(robot) {
-  // write code here
+  const newRobot = {};
+
+  for (const key in robot) {
+    if (newRobot.hasOwnProperty([robot[key]])) {
```
Comment (jstmpelowycz) [inline src/inverseRobot.js:30]: `robot[key]` looks like a new var

**Review by jstmpelowycz — CHANGES_REQUESTED** [review]

File: src/inverseRobot.js (hunk 4 above)
Comment (ik-web) [reply src/inverseRobot.js]: Fixed

**Review by jstmpelowycz — APPROVED** [review]
LGTM!

**Pull request closed without merging**